docs/core/v3.0.rst

```diff
@@ -0,0 +1,1762 @@
+of data between a variety of computing environments. The native byte
+order may differ between machines used to write and read the data.
+
+Each data type is associated with an identifier, which can be used in
+metadata documents to refer to the data type. For the data types
+defined in this specification, the identifier is a simple ASCII
+string. However, extensions may use any JSON value to identify a data
+type.
+
+
+Core data types
+---------------
+
+.. list-table:: Data types
+   :header-rows: 1
+
+   * - Identifier
+     - Numerical type
+     - Size (no. bytes)
+     - Byte order
+   * - ``bool``
+     - Boolean, with False encoded as ``\\x00`` and True encoded as ``\\x01``
+     - 1
+     - None
+   * - ``i1``
+     - signed integer
+     - 1
+     - None
+   * - ``<i2``
+     - signed integer
+     - 2
+     - little-endian
+   * - ``<i4``
+     - signed integer
+     - 4
+     - little-endian
+   * - ``<i8``
+     - signed integer
+     - 8
+     - little-endian
+   * - ``>i2``
+     - signed integer
+     - 2
+     - big-endian
+   * - ``>i4``
+     - signed integer
+     - 4
+     - big-endian
+   * - ``>i8``
+     - signed integer
+     - 8
+     - big-endian
+   * - ``u1``
+     - unsigned integer
+     - 1
+     - None
+   * - ``<u2``
+     - unsigned integer
+     - 2
+     - little-endian
+   * - ``<u4``
+     - unsigned integer
+     - 4
+     - little-endian
+   * - ``<u8``
+     - unsigned integer
+     - 8
+     - little-endian
+   * - ``>u2``
+     - unsigned integer
+     - 2
+     - big-endian
+   * - ``>u4``
+     - unsigned integer
+     - 4
+     - big-endian
+   * - ``>u8``
+     - unsigned integer
+     - 8
+     - big-endian
+   * - ``<f2``
+     - half precision float: sign bit, 5 bits exponent, 10 bits mantissa
+     - 2
+     - little-endian
+   * - ``<f4``
+     - single precision float: sign bit, 8 bits exponent, 23 bits mantissa
+     - 4
+     - little-endian
+   * - ``<f8``
+     - double precision float: sign bit, 11 bits exponent, 52 bits mantissa
+     - 8
+     - little-endian
+   * - ``>f2``
+     - half precision float: sign bit, 5 bits exponent, 10 bits mantissa
+     - 2
+     - big-endian
+   * - ``>f4``
+     - single precision float: sign bit, 8 bits exponent, 23 bits mantissa
+     - 4
+     - big-endian
+   * - ``>f8``
+     - double precision float: sign bit, 11 bits exponent, 52 bits mantissa
+     - 8
+     - big-endian
+   * - ``r*`` (Optional)
+     - raw bits,  use for extension type fallbacks
+     - variable, given by ``*``, is limited to be a multiple of 8.
+     - N/A
+
+
+Floating point types correspond to basic binary interchange formats as
+defined by IEEE 754-2008.
+
+Additionally to these base types, an implementation should also handle the
+raw/opaque pass-through type designated by the lower-case letter ``r`` followed
+by the number of bits, multiple of 8. For example, ``r8``, ``r16``, and ``r24``
+should be understood as fall-back types of respectively 1, 2, and 3 byte length.
```

Comment:
I think it would help to elaborate on `fall-back` here. As far as I understand the idea is that data stored by some extension would have e.g. the data type `r8`, if it has elements of size 8 bit. So a zarr implementation that does not implement the given extension could at least load the raw representation of the data.

Reply:
It would be more consistent if everything were in bits.

Reply:
Yep, fully agree @mkitti (as also discussed elsewhere.)

Reply:
I think this is resolved via #155, please also see #167.